.clang-tidy

```diff
@@ -0,0 +1,92 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+---
+Checks: '
```

Comment:
the checks preceded by `-` are disabled/ignored checks, the others are enabled. for instance:

* `-*` disables all defaults (this is good practice because it allows choosing your own style)
* `bugprone-*` enables all checks in the `bugprone` namespace (overwriting `-*`)
* `-bugprone-blabla` overwrites the previous statement for that specific check and disables it again